### PR TITLE
feat(asic-rs): use asic-rs components to help classify errors

### LINKED ISSUE #312: asicrs plugin loses per-chip / per-slot error detail because component_id is never populated
## Summary

The asicrs plugin emits every error with `component_id: None`, so the storage dedup key `(org_id, device_id, miner_error, component_id, component_type)` collides across logically-distinct components. Many real, distinct errors from a single miner upsert into one DB row, and the operator sees only the most recently written message.

## Repro

Whatsminer M30SPlusVH60 reported 20 raw errors in a single poll:

| Raw codes | Raw message |
|---|---|
| 264 | Power communication error. |
| 265, 266, 276, 277, 278, 279, 280 | Power unknown error. |
| 267 | Power watchdog protection. |
| 270 | Power input high voltage protection. |
| 271 | Power input low voltage protection. |
| 272 | Excessive power supply output warning. |
| 273 | Power input too high warning. |
| 274 | Power fan warning. |
| 320 | Slot 0 temperature reading error. |
| 551002, 551039, 551066, 551999 | Slot 1 chip {2, 39, 66, 999} has been reset. |
| 9110 | Unknown error type. |

Fleet's per-miner status modal showed only 5 rows:

1. Slot 1 chip 999 has been reset.
2. Power unknown error.
3. Excessive power supply output warning.
4. Power fan warning.
5. Unknown error type.

The four chip-reset events for chips 2, 39, 66, 999 plus the slot 0 temperature reading error all collapsed into row 1 because they share `(miner_error=VendorErrorUnmapped, component_type=HashBoard, component_id=NULL)` and chip 999 has the most recent timestamp.

## Root cause

- `plugin/asicrs/src/device.rs:513` hard-codes `component_id: None` in `to_device_errors()`.
- `plugin/asicrs/src/device.rs:1064-1133` (`classify_error()`) chooses `component_type` by keyword-matching the message text but never extracts the slot / chip identifier embedded in messages like `"Slot 1 chip 999 has been reset."`.
- `server/internal/domain/stores/sqlstores/error.go:72-93` upserts on `(org_id, device_id, miner_error, component_id, component_type)` and treats matches as the same error, overwriting `summary` and `last_seen_at`. Working as designed; the input is wrong.

## Proposed fix

In `plugin/asicrs/src/device.rs`:

1. In `classify_error()`, extract slot and chip identifiers from messages that mention them (e.g. regex `slot\s+(\d+)(?:\s+chip\s+(\d+))?` and similar for "board", "chain") and return them alongside the existing classification.
2. In `to_device_errors()`, populate `component_id` from those extracted identifiers using a canonical format (suggestion: `slot{N}` for slot-level, `slot{N}.chip{M}` for chip-level; `psu`, `fan{N}` for the non-hashboard cases).
3. Leave `component_id: None` only when the message genuinely has no sub-component identifier.

## Expected behavior after fix

For the repro miner above, the modal should show distinct rows for:
- Slot 1 chip 2 reset
- Slot 1 chip 39 reset
- Slot 1 chip 66 reset
- Slot 1 chip 999 reset
- Slot 0 temperature reading error

plus the existing PSU / fan / unknown rows, instead of all five collapsing to one chip-999 row.

## Out of scope

- Numeric-code -> canonical mapping table (currently keyword-only). Worth considering separately; would also break apart the 13 power codes currently fused into one "Power unknown error." row.
- Surfacing a "truncated" indicator in the UI when the server query paginates past 1000 per page. Distinct concern.

### PR DESCRIPTION
**Pull request by b-rowan**
Updates error classification to use the new `MinerComponent` to help classify errors when they come from asic-rs.

Also adds some tests for different edge cases in this new scenario, such as errors with or without components, and with or without error messages.

Fixes: #312

### CODE REVIEW

**Comment by b-rowan** [comment]
Worth mentioning this does lose some very slight granularity on error levels, namely, messages from asic-rs will no longer be classed as `Major`, only `Critical`, `Minor`, and `Info`.  Up for some discussion on this part, since this could possibly be reverted to parsing directly from the message strings.

File: plugin/asicrs/src/device.rs [hunk 1]
```diff
@@ -1060,38 +1061,64 @@ fn determine_board_status(board: &asic_rs_core::data::board::BoardData) -> pb::C
...
-    } else if lower.contains("control board") {
-        pb::MinerError::ControlBoardFailure
-    } else if lower.contains("firmware") {
-        pb::MinerError::FirmwareImageInvalid
-    } else {
-        pb::MinerError::VendorErrorUnmapped
+        return pb::MinerError::EepromReadFailure
+    }  else if lower.contains("firmware") {
+        return pb::MinerError::FirmwareImageInvalid
     };
```
[the remark this answers is not in the document]
Comment (b-rowan) [reply plugin/asicrs/src/device.rs]: Fixed.

File: plugin/asicrs/src/device.rs [hunk 2]
```diff
@@ -1060,29 +1061,48 @@ fn determine_board_status(board: &asic_rs_core::data::board::BoardData) -> pb::C
...
-        pb::MinerError::VendorErrorUnmapped
+        return pb::MinerError::FirmwareImageInvalid;
     };
 
-    let severity = if [
+    match m.component {
+        Some(MinerComponent::ControlBoard { .. }) => pb::MinerError::ControlBoardFailure,
+        Some(MinerComponent::HashBoard { chip_idx: None, .. }) => {
+            pb::MinerError::HashboardNotPresent
+        }
```
[the remark this answers is not in the document]
Comment (b-rowan) [reply plugin/asicrs/src/device.rs:1080]: Not sure about this, I suppose it depends on the error code, but likely not a better way to classify errors.  Can take a more in depth look into the different error types, but not sure if there is a better option here.

File: plugin/asicrs/src/device.rs [hunk 3]
```diff
@@ -1100,34 +1120,61 @@ fn classify_error(msg: &str) -> (pb::MinerError, pb::Severity, pb::ComponentType
     {
         pb::Severity::Minor
     } else {
-        pb::Severity::Major
-    };
+        match m.severity {
```
Comment (ankitgoswami) [inline plugin/asicrs/src/device.rs]: Thinking if this should be flipped where m.severity is checked first, then the keyword match is used to determine critical vs major?

File: plugin/asicrs/src/device.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ankitgoswami) [reply plugin/asicrs/src/device.rs:1080]: i think this isn't worse than before at least

File: plugin/asicrs/src/device.rs (hunk 3 above)
Comment (b-rowan) [reply plugin/asicrs/src/device.rs]: Yeah I think you're right, makes sense, I'll try to get this fixed.

**Review by mcharles-square — APPROVED** [review]

**Comment by mcharles-square** [comment]
Thanks for this @b-rowan I pushed a change to make the linter happy

File: plugin/asicrs/src/device.rs [hunk 4]
```diff
@@ -1060,74 +1061,116 @@ fn determine_board_status(board: &asic_rs_core::data::board::BoardData) -> pb::C
...
-        pb::MinerError::VendorErrorUnmapped
+        return pb::MinerError::FirmwareImageInvalid;
     };
 
-    let severity = if [
+    match m.component {
+        Some(MinerComponent::ControlBoard { .. }) => pb::MinerError::ControlBoardFailure,
+        Some(MinerComponent::HashBoard { chip_idx: None, .. }) => {
+            pb::MinerError::HashboardNotPresent
+        }
```
[the remark this answers is not in the document]
Comment (mcharles-square) [reply plugin/asicrs/src/device.rs:1080]: @ankitgoswami @b-rowan resolving this one to merge the PR but I think this should be addressed since it can actually be misleading. I put up a [draft PR](https://github.com/block/proto-fleet/pull/540) to address

[the remark this answers is not in the document]
Comment (b-rowan) [reply plugin/asicrs/src/device.rs:1080]: I think we would actually rather have some kind of generic hashboard error in theory, this is an error that can cause issues later, but probably should only be classified as a warning.  Might not be a bad idea to add warning levels to the classifications in asic-rs, since we do translations to Whatsminer error messages directly from the codes...
